be/src/exec/pipeline/exchange/sink_buffer.cpp

```diff
@@ -194,12 +194,39 @@ int64_t SinkBuffer::_network_time() {
 void SinkBuffer::cancel_one_sinker(RuntimeState* const state) {
     if (--_num_uncancelled_sinkers == 0) {
         _is_finishing = true;
+        if (state != nullptr && state->query_ctx() && state->query_ctx()->is_query_expired()) {
```

Comment:
state are never be nullptr.

Reply:
where guarantee this?

Reply:
RuntimeState is created in FragmentContext,  if here RuntimeState is nullptr, then the world-ending disaster would happen.

Reply:
sometime maybe not set in tests.
safety check.